sdk/src/main/java/com/mapbox/maps/ViewAnnotationManagerImpl.kt

```diff
@@ -178,13 +195,94 @@ internal class ViewAnnotationManagerImpl(
+    var north: Pair<ViewAnnotationOptions, Rect?>? = null
+    var east: Pair<ViewAnnotationOptions, Rect?>? = null
+    var west: Pair<ViewAnnotationOptions, Rect?>? = null
+    var south: Pair<ViewAnnotationOptions, Rect?>? = null
+
+    while (!isCorrectBound) {
+      val zoom = cameraOptionForCoordinates.zoom
+      isCorrectBound = true
+      viewAnnotationOptions.forEach { options ->
+        val frame = getViewAnnotationOptionsFrame(options)
+        val annotationBounds = calculateCoordinateBoundForAnnotation(options, zoom)
+        if (north == null || calculateCoordinateBoundForAnnotation(north!!.first, zoom).north() < annotationBounds.north()) {
+          north = Pair(options, frame)
+          isCorrectBound = false
+        }
+        if (east == null || calculateCoordinateBoundForAnnotation(east!!.first, zoom).east() < annotationBounds.east()) {
+          east = Pair(options, frame)
+          isCorrectBound = false
+        }
+        if (south == null || calculateCoordinateBoundForAnnotation(south!!.first, zoom).south() > annotationBounds.south()) {
+          south = Pair(options, frame)
+          isCorrectBound = false
+        }
+        if (west == null || calculateCoordinateBoundForAnnotation(west!!.first, zoom).west() > annotationBounds.west()) {
+          west = Pair(options, frame)
+          isCorrectBound = false
+        }
+      }
+
+      val coordinateBoundsForCamera = CoordinateBounds(
+        Point.fromLngLat((west!!.first.geometry as Point).longitude(), (south!!.first.geometry as Point).latitude()),
+        Point.fromLngLat((east!!.first.geometry as Point).longitude(), (north!!.first.geometry as Point).latitude())
+      )
+
+      val paddings = EdgeInsets(
+        (edgeInsets?.top ?: 0).toDouble().plus(abs((north!!.second?.top ?: 0.0).toDouble())),
+        (edgeInsets?.left ?: 0).toDouble().plus(abs((west!!.second?.left ?: 0.0).toDouble())),
+        (edgeInsets?.bottom ?: 0).toDouble().plus(abs((south!!.second?.bottom ?: 0.0).toDouble())),
+        (edgeInsets?.right ?: 0).toDouble().plus(abs((east!!.second?.right ?: 0.0).toDouble()))
+      )
+
+      cameraOptionForCoordinates = mapboxMap.cameraForCoordinateBounds(
+        coordinateBoundsForCamera,
+        paddings,
+        bearing,
+        pitch
+      )
+    }
+    return cameraOptionForCoordinates
+  }
+
+  /**
+   * Function to calculate coordinatebound associated with annotation option.
+   * this uses [MapboxMap.projectedMetersForCoordinate] and [MapboxMap.coordinateForProjectedMeters]
+   * function to get correct coordinate for northEast and southWest point of annotations.
+   */
+  @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+  internal fun calculateCoordinateBoundForAnnotation(
+    viewAnnotationOptions: ViewAnnotationOptions,
+    zoom: Double?
+  ): CoordinateBounds {
+    val frame = getViewAnnotationOptionsFrame(viewAnnotationOptions)
```

Comment:
can we create some default frame is can't get with `viewAnnotationOptions`
it will simplify lines 274-278